configd/src/apps/sentinel/service.cpp

```diff
@@ -18,6 +18,19 @@ extern sig_atomic_t stop;
 namespace config {
 namespace sentinel {
 
+namespace {
+
+std::string getTempDir() {
+    const char * configuredTmp = getenv("TMP");
```

Comment:
TMPDIR is likely wrong, but nothing sets TMP here. From a security perspective, we should use a directory with limited write permissions (i.e. owned by $VESPA_USER).